### PR TITLE
Added glue-qt package

### PR DESCRIPTION
**Pull request by astrofrog**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by astrofrog** [comment]
I am willing to be a maintainer of the package

**Comment by astrofrog** [comment]
@conda-forge/help-python - this is ready for review!

File: recipes/glue-qt/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    - echo >=0.6
+    - astropy >=4.0
+    - setuptools >=30.3.0
+    - qtpy >=1.9
+    - ipython >=4.0
+    - ipykernel >=4.0,!=5.0.0,!=5.1.0
+    - qtconsole >=4.3,!=5.4.2
+    - pvextractor >=0.2
+    - importlib-resources >=1.3  # [py<39]
+    - importlib-metadata >=3.6  # [py<310]
```
Comment (ocefpaf) [inline recipes/glue-qt/meta.yaml:39]: We have empty packages for these on modern Python. That way you can drop the selector and make this noarch.

Comment (astrofrog) [reply recipes/glue-qt/meta.yaml]: Thanks! Actually I just realised I need to add in python.app as a dependency which is OS-X only so can't actually specify noarch - but I've removed the selectors here.

File: recipes/glue-qt/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/glue-qt-{{ version }}.tar.gz
+  sha256: 26a7f10e8bb58426512921355a0b9078698170fcc66504b2174969e1d1073d42
+
+build:
+  skip: true  # [py<38]
```
Comment (ocefpaf) [inline recipes/glue-qt/meta.yaml:13]: I'm guessing there are no compiled parts, right? If so we can make this noarch.

```suggestion
  noarch: python
```

File: recipes/glue-qt/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+  entry_points:
+    - glue = glue_qt.main:main
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+  osx_is_app: true
+
+requirements:
+  build:
+    - python                                 # [build_platform != target_platform]
+    - cross-python_{{ target_platform }}     # [build_platform != target_platform]
```
Comment (ocefpaf) [inline recipes/glue-qt/meta.yaml:23]: If this can be noarch you won't need this.

```suggestion
```

File: recipes/glue-qt/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+  host:
+    - python
+    - setuptools-scm
+    - pip
+  run:
+
+    # The following is needed to make sure that the package works as a GUI
+    # application (glue needs to be run with python.app, not python)
+    - python.app  # [osx]
```
Comment (ocefpaf) [inline recipes/glue-qt/meta.yaml:32]: This is a trick to have an osx only dependency but keep it noarch.
See https://github.com/conda-forge/click-feedstock/blob/main/recipe/meta.yaml for an example.
(You'll probably need to adjust the build string too.)

```suggestion
    - python.app  # [osx]
    - __win      # [win]
    - __unix     # [unix]
```

**Comment by astrofrog** [comment]
@ocefpaf - thanks for the suggestions! Is there documentation regarding the build strings and what they should be for this kind of situation? Also is the linter error safe to ignore about the selectors?

File: recipes/glue-qt/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/glue-qt-{{ version }}.tar.gz
+  sha256: 26a7f10e8bb58426512921355a0b9078698170fcc66504b2174969e1d1073d42
+
+build:
+  noarch: python
+  entry_points:
+    - glue = glue_qt.main:main
+  script: {{ PYTHON }} -m pip install . -vv
```
Comment (ocefpaf) [inline recipes/glue-qt/meta.yaml:16]: I believe we also need this for the "semi-arch" package.

```suggestion
  script: {{ PYTHON }} -m pip install . -vv
  string: win_pyh{{ PKG_HASH }}_{{ PKG_BUILDNUM }}  # [win]
  string: unix_pyh{{ PKG_HASH }}_{{ PKG_BUILDNUM }}  # [unix]
```

**Comment by astrofrog** [comment]
Thanks for the remaining suggestion! I'll admit I don't quite understand the magic happening here for semi-noarch packages :sweat_smile:

**Comment by ocefpaf** [comment]
> @ocefpaf - thanks for the suggestions! Is there documentation regarding the build strings and what they should be for this kind of situation?

Those are not in our, see https://conda-forge.org/docs/maintainer/knowledge_base.html#noarch-packages-with-os-specific-dependencies

To be honest this is kind of new and I'm still getting used to it. Maybe we no longer need those. See https://github.com/conda-forge/conda-forge.github.io/pull/1839#discussion_r1008679066

> Also is the linter error safe to ignore about the selectors?

Yep. It is a bug and should be fixed by https://github.com/conda-forge/conda-smithy/pull/1735

**Pull request merged**